File: ant/src/com/appsecinc/ant/license/LICENSEHTM_LicenseFile.java
package com.appsecinc.ant.license;

public class LICENSEHTM_LicenseFile implements LicenseFile {

	@Override
	public String getExtension() {
		return "html";
	}

	@Override
	public boolean isMatch(String filename) {
		String filenameLowerCase = filename.toLowerCase();
		return filenameLowerCase.equals("license.html")
			|| filenameLowerCase.equals("license.htm") 
			|| filenameLowerCase.endsWith("-license.htm")
			|| filenameLowerCase.endsWith("-license.html");
	}

	@Override
	public String getSpec(String filename) {
		String filenameLowerCase = filename.toLowerCase();
		filenameLowerCase = filenameLowerCase.replace(".html", "");
		filenameLowerCase = filenameLowerCase.replace(".htm", "");
		filenameLowerCase = filenameLowerCase.replace("-license", "");
		filenameLowerCase = filenameLowerCase.replace("_license", "");
		if (filenameLowerCase.equals("license")) return "";
		return filenameLowerCase;
	}
	
	@Override
	public String getType() {
		return null;
	}
}

File: msbuild/src/AppSecInc.LicensesCollector/LICENSEHTM_LicenseFile.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace AppSecInc.LicensesCollector
{
    class LICENSEHTM_LicenseFile : ILicenseFile
    {
        #region ILicenseFile Members

        public string GetExtension()
        {
            return "html";
        }

        public bool IsMatch(string filename)
        {
            String filenameLowerCase = filename.ToLower();
            return filenameLowerCase.Equals("license.html")
                || filenameLowerCase.Equals("license.htm")
                || filenameLowerCase.EndsWith("-license.htm")
                || filenameLowerCase.EndsWith("-license.html");
        }

        public string GetSpec(string filename)
        {
            String filenameLowerCase = filename.ToLower();
            filenameLowerCase = filenameLowerCase.Replace(".html", "");
            filenameLowerCase = filenameLowerCase.Replace(".htm", "");
            filenameLowerCase = filenameLowerCase.Replace("-license", "");
            filenameLowerCase = filenameLowerCase.Replace("_license", "");
            if (filenameLowerCase.Equals("license")) return "";
            return filenameLowerCase;
        }

        public new string GetType()
        {
            return "HTML";
        }

        #endregion
    }
}
